# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/PFetch.ipdl
# Commit: 7a4e3f5f674a
# Full Hash: 7a4e3f5f674ad675fa084bf1f30a79a15e8c7d2a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r 9da00c1364a5 -r 7a4e3f5f674a dom/fetch/PFetch.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/fetch/PFetch.ipdl	Wed Jan 18 00:50:19 2023 +0000
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include protocol PBackground;
+
+include ClientIPCTypes;
+include FetchTypes;
+include IPCServiceWorkerDescriptor;
+include NeckoChannelParams;
+include PBackgroundSharedTypes;
+include PerformanceTimingTypes;
+
+using FetchDriverObserver::EndReason from "mozilla/dom/FetchDriver.h";
+
+namespace mozilla {
+namespace dom {
+
+struct FetchOpArgs{
+  IPCInternalRequest request;
+  PrincipalInfo principalInfo;
+  nsCString workerScript;
+  IPCClientInfo clientInfo;
+  IPCServiceWorkerDescriptor? controller;
+  CookieJarSettingsArgs? cookieJarSettings;
+  bool needOnDataAvailable;
+  bool hasCSPEventListener;
+};
+
+protocol PFetch {
+  manager PBackground;
+
+  parent:
+    async FetchOp(FetchOpArgs aArgs);
+
+    async AbortFetchOp();
+
+  child:
+    async OnResponseAvailableInternal(ParentToChildInternalResponse aResponse);
+
+    async OnResponseEnd(ResponseEndArgs aResponseEndArgs);
+
+    async OnDataAvailable();
+
+    async OnFlushConsoleReport(ConsoleReportCollected[] aReports);
+
+    async OnCSPViolationEvent(nsString aJSON);
+
+    async __delete__(nsresult aResult);
+};
+
+}
+}